From 53e66aea4c1b3617363fd138a7903190d964da6b Mon Sep 17 00:00:00 2001
From: Tom치코 Chv치tal <tchvatal@suse.cz>
Date: Fri, 30 Sep 2011 09:51:04 +0000
Subject: This is merge of f6f8f67460bba296c49f794d14621b7313df4f7d.

It is not posible to git-am due to merge to core/ repository.
It had one minor collision on pdfioutdev_gpl.hxx.

Orginal header:
From f6f8f67460bba296c49f794d14621b7313df4f7d Mon Sep 17 00:00:00 2001
From: Caol치n McNamara <caolanm@redhat.com>
Date: Tue, 19 Jul 2011 07:54:07 +0000
Subject: poppler 0.17.0 changed its api for no some reason or other
---
diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index f7741b4..5022dd5 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -489,7 +489,11 @@ void PDFOutDev::endPage()
     printf("endPage\n");
 }
 
+#if POPPLER_CHECK_VERSION(0, 17, 0)
+void PDFOutDev::processLink(AnnotLink *link, Catalog *catalog)
+#else
 void PDFOutDev::processLink(Link* link, Catalog*)
+#endif
 {
     assert(link);
 
diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
index 3ac0f49..8a1708c 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
@@ -65,7 +65,10 @@ class GfxPath;
 class GfxFont;
 class PDFDoc;
 #ifndef SYSTEM_POPPLER
+#define POPPLER_CHECK_VERSION(major,minor,micro) (0)
 typedef GString GooString;
+#else
+#include <glib/poppler-features.h>
 #endif
 
 namespace pdfi
@@ -193,9 +196,13 @@ namespace pdfi
         // Convert between device and user coordinates.
         // virtual void cvtDevToUser(double dx, double dy, double *ux, double *uy);
         // virtual void cvtUserToDev(double ux, double uy, int *dx, int *dy);
-        
+       
+        #if POPPLER_CHECK_VERSION(0, 17, 0)
+        virtual void processLink(AnnotLink *link, Catalog *catalog);
+        #else
         //----- link borders
         virtual void processLink(Link *link, Catalog *catalog);
+        #endif 
         
         //----- save/restore graphics state
         virtual void saveState(GfxState *state);
--
cgit v0.9.0.2-2-gbebe
